### PR TITLE
Removed Firefox buildid from os.js

### LINKED ISSUE #3936: Remove the Firefox buildid string matching from os.js
This causes a large uptick in the filesize of obfuscated os.js since it adds something like 600 strings, many of which are 10+ chars. This can be crucial when serving the detection script to mobile devices, as the parse/interpret overhead of calculating hundreds of strings becomes noticeable (not to mention network latency).

I don't think any exploits depend on these, and I don't see why you'd want to use this instead of feature detection, which will pin other Gecko browsers. It's not a bad technique, it's just rather expensive on resources. Additionally, if we did want to keep this technique, it would make sense to move all the strings over to Ruby, and just pass the client's `navigator.buildid` back, instead of sending 600 strings over to the client.

### PR DESCRIPTION
**Pull request by MSadek-r7**
Closes #3936
# Verification

Copy and paste os.js in the following format:

``` html
<html>
<script>
.... os.js code here ...

var osInfo = os_detect.getVersion();
alert(osInfo.os_name);
</script>
</html>
```
- [ ] Save the HTML file as /tmp/test.html
- [ ] Under /tmp, do this in a terminal: ruby -run -e httpd . -p 8181
- [ ] On the Windows box, open IE, and then open the test.html
- [ ] You should see a prompt with the proper operating system

### CODE REVIEW

File: data/js/detect/os.js [hunk 1]
```diff
@@ -343,401 +343,8 @@ os_detect.getVersion = function(){
 		if (version.match(/Linux/)) {
 			os_name = oses_linux;
 		}
-		// end navigator.oscpu checks
```
Comment (wvu-r7) [inline data/js/detect/os.js]: Remove this newline?

**Comment by wvu-r7** [comment]
Who wants this?

**Comment by jvazquez-r7** [comment]
In order to complete this PR, the next consideration should be take into account:

```
Additionally, if we did want to keep this technique, it would make sense to move all the strings over to Ruby, and just pass the client's navigator.buildid back, instead of sending 600 strings over to the client.
```

I think worths to keep this technique in the proposed way, instead of just deleting it. See #3936 for additional feedback. If there is agreement on just delete the technique, then it's good to go :) thanks @MSadek-r7 !

**Comment by joevennix** [comment]
@jvazquez-r7, imho the buildid detection seems more trouble than it's worth. It's very accurate, which is awesome, but is painful to maintain (which is why it's not updated since 2012), and without hardcoding every buildid ever released this technique will eventually miss some versions. If we have a module that _needs_ this level of granularity, I would say that is an uncommon case and the module should implement this itself across its supported version range; this could be achieved by having BES report the `buildid` value up to the module.

File: data/js/detect/os.js [hunk 2]
```diff
@@ -344,400 +343,7 @@ os_detect.getVersion = function(){
...
-				}
-				break;
-		}
-		//if (ua_is_lying) { alert("UA is lying"); }
-		//alert(ua_version + " vs " + navigator.userAgent);
-
-		// end navigator.buildID checks
-
-	} else if (typeof ScriptEngineMajorVersion == "function") {
+		else if (typeof ScriptEngineMajorVersion == "function") {
```
Comment (wvu-r7) [inline data/js/detect/os.js]: You are missing a brace here.

**Pull request closed without merging**